numpy/core/src/_simd/_simd_inc.h.src

```diff
@@ -0,0 +1,257 @@
+simd_data_is_sequence(simd_data_type dtype);
+// returns 1 if dtype is a vector, otherwise '0'
+static int
+simd_data_is_vector(simd_data_type dtype);
+// returns 1 if dtype is a multi-vector x2, otherwise '0'
+static int
+simd_data_is_vectorx2(simd_data_type dtype);
+// returns 1 if dtype is a multi-vector x3, otherwise '0'
+static int
+simd_data_is_vectorx3(simd_data_type dtype);
+/**
+ * converts dtype to equivalent vector type,
+ * e.g. u8 -> vu8
+ */
+static simd_data_type
+simd_data_cvt2vector(simd_data_type dtype);
+/**
+ * converts dtype to equivalent scalar type,
+ * e.g. vu8 -> u8
+ */
+static simd_data_type
+simd_data_cvt2scalar(simd_data_type dtype);
+
+/************************************
+ ** Declarations (inc_convert)
+ ************************************/
+/**
+ * convert PyObject to simd_data, raise Python exception on
+ * failure.
+ * NOTE: simd_obj2data_clear() must be called after get done
+ * from returned data.
+ * example:
+ ** simd_data d = simd_obj2data(obj, simd_data_qu8);
+ ** ...
+ ** simd_obj2data_clear(d, simd_data_qu8);
+ */
+static simd_data
+simd_obj2data(PyObject *obj, simd_data_type dtype);
+/**
+ * convert simd_data to PyObject, raise Python exception on
+ * failure and return NULL.
+ */
+static PyObject *
+simd_data2obj(simd_data data, simd_data_type dtype);
+
+static int
+simd_data2seq(PyObject *seq, simd_data data, simd_data_type dtype);
+/**
+ * free memory allocations, that caused by simd_obj2data,
+ * mainly to free sequences pointer.
+ */
+static void
+simd_obj2data_clear(simd_data data, simd_data_type dtype);
+
+/************************************
+ ** Declarations (inc_vector)
+ ************************************/
+/**
+ * convert simd_data to PyObject(simd_vector),
+ * raise Python exception on failure and returns NULL.
+ */
+static simd_vector *
+simd_data2vector(simd_data data, simd_data_type type);
+/**
+ * convert PyObject(simd_vector) to simd_data,
+ * raise Python exception on failure.
+ */
+static simd_data
+simd_vector2data(simd_vector *vec, simd_data_type type);
+/**
+ * init and register vector type(PyTypeObject) to PyModule,
+ * vector type can be reached through attribute 'vector_type'.
+ * return 0 on error, 1 on success.
+ */
+static int
+simd_vector_register(PyObject *module);
+/************************************
+ ** Declarations (inc_arg)
+ ************************************/
+/**
+ * Parse and convert intrinsic arguments to SIMD data types,
+ * returns false on failure.
+ *
+ * py_args : module's method arguments
+ * args : array of 'simd_arg'
+ * args_n : length of 'args'
+ */
+static int
+simd_arg_parse(PyObject *py_args, simd_arg *args, int args_n);
+
+static void
+simd_arg_clear(simd_arg *args, int args_n);
+
+/************************************
+ ** Include the definitions
+ ************************************/
+#include "_simd_inc_data.h"
+#include "_simd_inc_convert.h"
+#include "_simd_inc_vector.h"
+#include "_simd_inc_arg.h"
+#include "_simd_inc_easyintrin.h"
```

Comment:
Rather than including these files, could you invert that: turn them into C files that include this one?

Reply:
I moved them to the disapatch-able source.

Reply:
done